src/controllers/csi/provisioner/controller.go

```diff
@@ -89,72 +122,110 @@ func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request r
 		log.Info("CSI driver not needed")
 		return reconcile.Result{RequeueAfter: longRequeueDuration}, nil
 	}
-
-	if dk.ConnectionInfo().TenantUUID == "" {
+	tenant := dk.ConnectionInfo().TenantUUID
+	if tenant == "" {
 		log.Info("DynaKube instance has not been reconciled yet and some values usually cached are missing, retrying in a few seconds")
 		return reconcile.Result{RequeueAfter: shortRequeueDuration}, nil
 	}
 
+	if provisioner.parallelDownloadsEnabled() {
+		if provisioner.parallelDownloadsLimitReached() {
+			log.Info("max parallel downloads reached, requeued")
+			return reconcile.Result{RequeueAfter: shortRequeueDuration}, nil
+		}
+		if provisioner.isCurrentlyDownloading(tenant) {
+			log.Info("still downloading", "tenant", tenant)
```

Comment:
nit: it should be able to cancel download it it's stalled using `context.Context`, but that should be done in another PR

Reply:
that would require that the download would care for that context but it doesn't (that part of the code was written before my time sooo 😬 )